src/app/cmd/copy_cel.cpp

```diff
@@ -116,12 +116,18 @@ void CopyCel::onExecute()
   }
   // For transparent layers
   else {
-    if (dstCel)
+    const bool isDstCelEmpty = bool(!dstCel);
+    int dstCelZOrder = (srcCel ? srcCel->zIndex() : 0);
+    if (dstCel) {
+      dstCelZOrder = dstCel->zIndex();
       executeAndAdd(new cmd::RemoveCel(dstCel));
+    }
 
     if (srcCel) {
-      if (createLink)
+      if (createLink) {
         dstCel = Cel::MakeLink(m_dstFrame, srcCel);
+        dstCel->setZIndex(dstCelZOrder);
+      }
```

Comment:
This `dstCelZOrder` is not required if we keep the change in `Cel::MakeLink()` as it was, because:
1. `dstCelZOrder` is used only when `srcCel != nullptr`
2. so `dstCelZOrder = srcCel->zIndex()`
3. and `dstCelZOrder` is used only in `dstCel->setZIndex(dstCelZOrder);` after `dstCel = Cel::MakeLink(m_dstFrame, srcCel);`

So if we set the zindex in `MakeLink()` (as we was doing), we don't require this change at all.